src/plugins/intel_gpu/src/graph/network.cpp

```diff
@@ -688,6 +692,39 @@ void network::add_default_output_chains() {
     }
 }
 
+void network::calculate_weights_cache_capacity() {
+    auto get_weights_size = [](const program_node& node) {
+        size_t weights_size = 0;
+        if (node.is_type<fully_connected>()) {
+            weights_size = node.as<fully_connected>().weights().get_output_layout().bytes_count();
+        } else if (node.is_type<convolution>()) {
+            weights_size = node.as<convolution>().weights().get_output_layout().bytes_count();
+        } else if (node.is_type<deconvolution>()) {
+            weights_size = node.as<deconvolution>().weights().get_output_layout().bytes_count();
+        }
+
+        return weights_size;
+    };
+
+    size_t total_weights_size = 0;
+    for (auto node : _program->get_processing_order())
+        total_weights_size += get_weights_size(*node);
+
+    total_weights_size *= _config.get_property(ov::streams::num);
+
+    if (total_weights_size != 0) {
+        const size_t required_weights_cache_capacity = 3;
+        const size_t max_device_mem_size = _engine.get_device_info().max_global_mem_size;
+        const size_t max_weights_cache_capacity = max_device_mem_size / total_weights_size;
```

Comment:
Ideally, this logic should respect memory required for activations and memory for other constants